pypeman/contrib/http.py

```diff
@@ -1,33 +1,82 @@
 import asyncio
 import ssl
 import warnings
-
-from pypeman import endpoints, channels, nodes, message
+import socket
 
 import aiohttp
 from aiohttp import web
 
+from pypeman import endpoints, channels, nodes, message
+from pypeman.errors import PypemanParamError
+
+
 class HTTPEndpoint(endpoints.BaseEndpoint):
     """
     Endpoint to receive HTTP connection from outside.
 
-    :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
-        `client_max_size`
     """
 
-    def __init__(self, adress=None, address='127.0.0.1', port='8080', loop=None, http_args=None):
+    def __init__(
+            self, adress=None, 
+            address=None, 
+            port=None, 
+            loop=None, 
+            http_args=None, 
+            url=None, 
+            sock=None,
+            reuse_port=False,
+            ):
+        """
+            :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
```

Comment:
Can you describe `url` param usage ?

Reply:
Also i'm not sure but i think that "X.X.X.X:PORT" isn't enough to be named URL.

Reply:
perhaps we should find another name than url. (perhaps socketstr?)

It's just a string describing the listening socket, which can be either:
a TCP socket of format hostname:port 
or a unix domain socket of type pointing to the socket descriptor on the file system

Reply:
Yes another name but i prefer `host`.

Reply:
I think ideally one environment variable should be sufficient to describe any potential way a server can listen to an incoming request, which means be able to  describe either a unix or a TCP socket.

For users that don't feel at ease with a socket parameter we could leave (without obsoleting them) the address / port parameters, which are perfectly describing a TCP socket, but would require two environment variables to be passed to settings.

The advantage of the name socket is, that it really describes what we talk about,
as what we listen to is either a unix socket or a tcp socket.  

We do not listen to a host, but on one of the current host's unix sockets or on a TCP port on a certain address (interface) of the current host.

socket is also the name, that is used for example by the uwsgi server (https://uwsgi-docs.readthedocs.io/en/latest/Options.html#socket)

We could support following syntaxes
socket = <socket_object> # an already bound socket object
socket = :8080  (for 127.0.0.1:8080) (probably the most often used for simple applications)
socket = 127.0.0.1:8080
socket = /path/to/socket_file (probably most often used larger projects with a reverse proxy for access control, as unix sockets have less overhead than TCP sockets and can be protected with file permissions)


I just don't have a good idea how how to have an easy syntax, that would allow setting reuse_port with the same environment variable, that requires the rest of the socket, so probably, so perhaps we can live with two env vars

Reply:
I think the most common and obvious use case is to use an address and port to configure a server. That is why the name "socket", even if it actually corresponds to this use, bothers me.

"address"," host","port","server" seem to me to be clear names whose values are simple to understand, whereas for "socket" I have the feeling that it is necessary to provide a "socket"object.

Can we keep a `server` parameter next to the `socket` parameter?

Usages can be different (and probably exclusive ?):
- "server" handle the syntaxes : <ip>, <:port> and <ip:port>
- and "socket" handle : <socket object> and <path/to/socket>

Reply:
better ?

Reply:
Fine, Thanks !